cypress/tests/data-source.spec.ts

```diff
@@ -0,0 +1,159 @@
+  Providers,
+  TEST_DATA_SOURCE,
+} from '../constants/tests';
+import { app } from '../views/common';
+import {
+  checkDataSourceCreation,
+  createDataSource,
+  createFormInputValidation,
+  deleteDataSourceResources,
+  deleteUsingDetailsPageKebabMenu,
+  navigateToCreatePage,
+  navigateToListPageViaBreadCrumbs,
+} from '../views/data-resource';
+import { MCGMSCommon } from '../views/mcg-ms-common';
+
+describe('tests for data source creation', () => {
+  before(() => {
+    cy.login();
+  });
+
+  beforeEach(() => {
+    MCGMSCommon.visitDataSourceListPage();
+    navigateToCreatePage();
+    app.waitForLoad();
+  });
+
+  afterEach(() => {
+    deleteUsingDetailsPageKebabMenu(TEST_DATA_SOURCE);
+    deleteDataSourceResources(TEST_DATA_SOURCE, DATA_FEDERATION_NAMESPACE);
+  });
+
+  after(() => {
+    cy.logout();
+  });
+
+  it('creates a data source having AWS as the provider', () => {
+    createDataSource(Providers.AWS, TEST_DATA_SOURCE);
+    checkDataSourceCreation(TEST_DATA_SOURCE, DATA_FEDERATION_NAMESPACE);
+  });
+
+  it('creates a data source having S3 Compatible as the provider', () => {
+    createDataSource(Providers.S3, TEST_DATA_SOURCE);
+    checkDataSourceCreation(TEST_DATA_SOURCE, DATA_FEDERATION_NAMESPACE);
+  });
+
+  it('creates a data source having Azure Blob as the provider', () => {
+    createDataSource(Providers.AZURE, TEST_DATA_SOURCE);
+    checkDataSourceCreation(TEST_DATA_SOURCE, DATA_FEDERATION_NAMESPACE);
+  });
+
+  it('creates a data source having IBM COS as the provider', () => {
+    createDataSource(Providers.IBM, TEST_DATA_SOURCE);
+    checkDataSourceCreation(TEST_DATA_SOURCE, DATA_FEDERATION_NAMESPACE);
+  });
+});
+
+describe('tests for data source deletion', () => {
+  before(() => {
+    cy.login();
+  });
+
+  beforeEach(() => {
+    MCGMSCommon.visitDataSourceListPage();
+    navigateToCreatePage();
+    app.waitForLoad();
+    createDataSource(Providers.AWS, TEST_DATA_SOURCE);
+    checkDataSourceCreation(TEST_DATA_SOURCE, DATA_FEDERATION_NAMESPACE, false);
+  });
+
+  afterEach(() => {
+    deleteDataSourceResources(TEST_DATA_SOURCE, DATA_FEDERATION_NAMESPACE);
+  });
+
+  after(() => {
+    cy.logout();
+  });
+
+  it('checks the data source is successfully deleted from the details page', () => {
+    deleteUsingDetailsPageKebabMenu(TEST_DATA_SOURCE);
+  });
+
+  it('checks the data source is successfully deleted from the list page', () => {
+    navigateToListPageViaBreadCrumbs(TEST_DATA_SOURCE);
+    cy.log(`deleting ${TEST_DATA_SOURCE} from the list page using kebab menu`);
+    cy.byTestRows('resource-row')
+      .should('contain', TEST_DATA_SOURCE)
+      .byTestID(`${TEST_DATA_SOURCE}-kebab`)
+      .click();
+    cy.byTestDropDownMenu('delete-data-source').click();
+    cy.byTestID('delete-action').click();
+    cy.byTestRows('resource-row', { timeout: 2 * MINUTE }).should('not.exist'); // kept it 2 minute as sometimes the list page takes times to reflect the changes
+  });
+});
```

Comment:
why do we need this again ?
In previous `describe('tests for data source creation'` we are already testing the deletion as well... why do it all again ?

Reply:
those deletion tests are hidden after each, wanted to have a standalone test for deletion

Reply:
that I can see as well... still we are using same functions in both describe, ultimately we are repeating the test checks...

Reply:
what's your opinion on this then?

Reply:
I have mixed views on this... there is repetition, but not of code, repetition of test cases (creating and deleting again which we are already doing in first describe as well)... 
to me it seems like we can incorporate deletion test cases in first describe itself (maybe in afterEach or separate it block)...
we can do whatever @alfonsomthd opinion is on this ??

Reply:
sure, my opinion is to keep the tests separate and independent. 
for deletion of resource of the creation tests, we can do it by any means like via cmd or UI whichever is faster, smaller, and easier.
whereas for particular deletion tests, we need to do it via UI.
in this case for lesser code and ease of doing I used the same function for both which serves both scenarios but makes it looks repetitive.
we can do whatever @alfonsomthd  opinion is on this ??

Reply:
From my understanding:
* We have a GUI test for covering the data source deletion.
* The aforementioned method is reused (no code duplication) for doing deletion/cleanup after each creation test because of the desire of keeping tests isolated.

We can move forward with the current approach,
but for the sake of performance, in the future we could refactor it so instead of deleting data source after each test, we could follow this [recommendation](https://docs.cypress.io/guides/references/best-practices#Using-after-or-afterEach-hooks) and:
1. ensure that all resources with unique names (i.e. "data-source-azure, data-source-aws, ...") are deleted in an `before` hook.
2. Run creation tests (they will not collide as they'll have unique names, no need to erase them every time).
3. Run a GUI deletion test over one of the created ones.